components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/internal/RoleManagementServiceImpl.java

```diff
@@ -357,4 +359,31 @@ private String getAuditData(String tenantDomain, String newRoleName) {
 
         return (String.format("Tenant Domain : %s, New Role Name : %s", tenantDomain, newRoleName));
     }
+
+    /**
+     * Get the initiator for audit logs.
+     *
+     * @param tenantDomain Tenant domain.
+     * @return initiator based on whether log masking is enabled or not.
+     */
+    private String getInitiator(String tenantDomain) {
+
+        String user = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        String initiator = null;
+        if (LoggerUtils.isLogMaskingEnable) {
+            if (StringUtils.isNotBlank(user) && StringUtils.isNotBlank(tenantDomain)) {
+                initiator = IdentityUtil.getInitiatorId(user, tenantDomain);
+                if (StringUtils.isNotBlank(initiator)) {
+                    return initiator;
+                }
+            }
+            if (StringUtils.isNotBlank(user)) {
+                return LoggerUtils.getMaskedContent(UserCoreUtil.addTenantDomainToEntry(user, tenantDomain));
+            }
+            return LoggerUtils.getMaskedContent(CarbonConstants.REGISTRY_SYSTEM_USERNAME);
+        } else if (StringUtils.isNotBlank(user)) {
+                return UserCoreUtil.addTenantDomainToEntry(user, tenantDomain);
+        }
+        return CarbonConstants.REGISTRY_SYSTEM_USERNAME;
+    }
```

Comment:
is this method duplicated in multiple places?

Reply:
moved to LoggerUtils class